src/xo-app/new/sr/index.js

```diff
@@ -0,0 +1,621 @@
+    } = this.state
+
+    try {
+      if (type === 'nfs' || type === 'nfsiso') {
+        const paths = await probeSrNfs(host, server.value)
+        this.setState({
+          usage: undefined,
+          paths
+        })
+      } else if (type === 'iscsi') {
+        const iqns = await probeSrIscsiIqns(host, server.value, port.value, username && username.value, password && password.value)
+        if (!iqns.length) {
+          info('iSCSI Detection', 'No IQNs found')
+        } else {
+          this.setState({
+            usage: undefined,
+            iqns
+          })
+        }
+      }
+    } catch (err) {
+      error('Server Detection', err.message || String(err))
+    }
+  }
+
+  _handleSrPathSelection = async path => {
+    const {
+      server
+    } = this.refs
+    const {
+      host
+    } = this.state
+
+    try {
+      this.setState({loading: true})
+      const list = await probeSrNfsExists(host, server.value, path)
+      const srIds = map(this.getHostSrs(), sr => sr.id)
+      const used = filter(list, item => includes(srIds, item.id))
+      const unused = filter(list, item => !includes(srIds, item.id))
+      this.setState({
+        path,
+        usage: true,
+        used,
+        unused,
+        summary: used.length <= 0
+      })
+    } catch (err) {
+      error('NFS Error', err.message || String(err))
+    } finally {
+      this.setState({loading: undefined})
+    }
+  }
+
+  _reattach = async uuid => {
+    const {
+      host,
+      type
+    } = this.state
+
+    let {
+      name,
+      description
+    } = this.refs
+
+    name = trim(name)
+    description = trim(description)
+    if (isEmpty(name) || isEmpty(description)) {
+      error('Missing General Parameters', 'Please complete General Information')
+    }
+
+    const method = (type === 'nfsiso') ? reattachSrIso : reattachSr
+    try {
+      await method(host, uuid, name, description, type)
+    } catch (err) {
+      error('Reattach', err.message || String(err))
+    }
+  }
+
+  render () {
+    const { hosts } = this.props
+    const {
+      auth,
+      host,
+      iqns,
+      loading,
+      lockCreation,
+      lun,
+      luns,
+      path,
+      paths,
+      summary,
+      type,
+      unused,
+      usage,
+      used
+    } = this.state
+    const { formatMessage } = this.props.intl
+
+    return (
+      <form id='newSrForm'>
+        <Wizard>
+          <Section icon='sr' title='newSrGeneral'>
+            <fieldset className='form-group'>
+              <label>{_('newSrHost')}</label>
+              <SelectHost
+                options={hosts}
+                onChange={this._handleSrHostSelection}
+              />
+              <label htmlFor='srName'>{_('newSrName')}</label>
+              <input
+                id='srName'
+                className='form-control'
+                placeholder='storage name'
+                ref='name'
+                onBlur={event => { this._handleNameChange(event.target.value) }}
```

Comment:
Avoid creating function in `render()`, make `_handleNameChange()` accept an event instead.